http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/RuntimeRoutes.scala

```diff
@@ -277,37 +300,21 @@ object RuntimeRoutes {
         case CloudService.Dataproc =>
           x.as[RuntimeConfigRequest.DataprocConfig]
         case CloudService.GCE =>
-          x.as[RuntimeConfigRequest.GceConfig]
+          for {
+            machineType <- x.downField("machineType").as[Option[MachineTypeName]]
+            pd <- x.downField("persistentDisk").as[Option[PersistentDiskRequest]]
+            res <- pd match {
```

Comment:
OK, this logic is key.. if `cloudService == GCE` in the request and it _has_ a `persistentDisk` then it's a `GceWithPdConfig`. If `cloudService == GCE` and it _doesn't_ have a `persistentDisk` then it's a `GceConfig`

Reply:
yea, in request that's the case...in decoder for `RuntimeConfig`, it'll be a bit different, we'll look at `diskSize` field instead because `diskSize` is required for `RuntimeConfig.GceConfig`, while `persistentDiskId` in `RuntimeConfig.GceWithPd` is optional